corehq/apps/userreports/tasks.py

```diff
@@ -80,15 +83,22 @@ def rebuild_indicators(indicator_config_id):
         redis_key = _get_redis_key_for_config(config)
 
     adapter.rebuild_table()
-    relevant_ids = get_doc_ids_in_domain_by_type(
-        config.domain,
-        config.referenced_doc_type,
-        database=couchdb,
-    )
-    for docs in chunked(relevant_ids, 1000):
-        redis_client.sadd(redis_key, *docs)
-
-    _build_indicators(indicator_config_id, relevant_ids)
+    view_kwargs = {
+        'reduce': False,
+        'startkey': [config.domain, config.referenced_doc_type],
+        'include_docs': False
+    }
+    relevant_ids_chunk = []
+    for relevant_id in paginate_view(
+            couchdb,
+            'by_domain_doc_type_date/view',
```

Comment:
this is a little scary since the view itself is definitely not guaranteed to meet the recommended criteria for `paginate_view` (from doc string). however, forms, cases and users do _mostly_ have dates in that view so it may be acceptable (cases have a phone date which is unfortunate):

```
    Note: If the contents of the couch view do not change over the duration of
    the paginate_view call, this is guaranteed to have the same results
    as a direct view call. If the view is updated, however,
    paginate_views may skip docs that were added/updated during this period
    or may include docs that were removed/updated during this period.
    For this reason, it's best to use this with views that update infrequently
    or that are sorted by date modified and/or add-only,
    or when exactness is not a strict requirement
```